lib/ansible/modules/crypto/openssl_cert.py

```diff
@@ -0,0 +1,996 @@
+        except:
+            e = get_exception()
+            raise CertificateError(e)
+        cert.sign(self.privatekey, self.selfsigned_digest)
+        self.certificate = cert
+
+        backupdest = ""
+        if self.changed and not self.module.check_mode:
+            if self.backup and os.path.exists(self.dest):
+                backupdest = module.backup_local(self.dest)
+            cert_file = open(self.dest, 'w')
+            cert_file.write(crypto.dump_certificate(crypto.FILETYPE_PEM, self.certificate))
+            cert_file.close()
+
+    def dump(self):
+
+        result = {
+            'changed': self.changed,
+            'cert_path': self.dest,
+            'notBefore': self.notBefore,
+            'notAfter': self.notAfter,
+            'serial_number': self.serial_number,
+            'backup': backupdest,
+            'warnings': self.warnings,
+        }
+
+        return result
+
+
+class AssertOnlyCertificate(Certificate):
+    '''validate the supplied certificate.'''
+
+    def __init__(self, module):
+        Certificate.__init__(self, module)
+
+    def generate(self):
+        '''Don't generate anything'''
+        pass
+
+    def dump(self):
+
+        result = {
+            'changed': self.changed,
+            'cert_path': self.dest,
+            'warnings': self.warnings,
+        }
+
+        return result
+
+
+class LetsEncryptCertificate(Certificate):
+    '''Retrieve Let's Encrypt certificate.'''
+
+    def __init__(self, module):
+        Certificate.__init__(self, module)
+        self.accountkey = module.params['letsencrypt_accountkey']
+        self.challenge_path = module.params['letsencrypt_challenge']
+
+    def generate(self):
+        if not self.force and os.path.exists(self.dest):
+            self.changed = False
+            return
+
+        backupdest = ""
+        if self.changed and not self.module.check_mode:
+            if self.backup and os.path.exists(self.dest):
+                backupdest = module.backup_local(self.dest)
+
+            # TODO (spredzy): Ugly part should be done directly by interacting
+            # with the acme protocol through python-acme
+            try:
+                os.system('acme-tiny --account-key %s --csr %s --acme-dir %s > %s' % (self.accountkey, self.csr_path, self.challenge_path, self.dest))
+            except:
+                e = get_exception()
+                raise CertificateError(e)
+
+    def dump(self):
+
+        result = {
+            'changed': self.changed,
+            'cert_path': self.dest,
+            'account_key': self.accountkey,
+            'backup': backupdest,
+            'warnings': self.warnings,
+        }
+
+        return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            dest=dict(required=True, aliases=['path', 'destfile'], type='path'),
```

Comment:
Replace `dest` by `path` to be consistent with the other modules (openssl_privatekey, openssl_publickey, openssl_csr)